### PR TITLE
Restructure the whole directory structure based on other contributors feedback.

### PR DESCRIPTION
**Pull request by tanvir-retailai**
1. framework folder
2. middlewares folder
3. packages folder

### CODE REVIEW

**Review by gaoweiqiao-tre — APPROVED** [review]
LGTM

File: framework/bootstrap/kernel.go [hunk 1]
```diff
@@ -51,11 +51,19 @@ func New() *echo.Echo {
 	// Hide default `Echo` banner during startup.
 	e.HideBanner = true
 
+	// XXX: - IMPORTANT
+
+	// !!! Do not use the original echo context in the goroutine since it will be released after the handler returned.
+	// !!! https://github.com/labstack/echo/issues/1633
```
Comment (yu-yk) [inline framework/bootstrap/kernel.go:57]: not clean?

**Review by yu-yk — CHANGES_REQUESTED** [review]

File: framework/bootstrap/kernel.go [hunk 2]
```diff
@@ -51,11 +51,19 @@ func New() *echo.Echo {
...
+
 	// Set viper path and read configuration. You must keep `env.json` file in the root of your project.
 	viper.AddConfigPath(".")
 	viper.SetConfigType("json")
 	viper.SetConfigName("env")
 
+	// !!! Do not use the original echo context in the goroutine since it will be released after the handler returned.
+	// !!! https://github.com/labstack/echo/issues/1633
+
```
Comment (SunJinping-TRE-C) [inline framework/bootstrap/kernel.go:66]: here also need clean

File: framework/bootstrap/kernel.go [hunk 3]
```diff
@@ -65,6 +73,12 @@ func New() *echo.Echo {
 		os.Exit(1)
 	}
 
+	// IMPORTANT:
+	// DEBUG:
+
+	// !!! Do not use the original echo context in the goroutine since it will be released after the handler returned.
+	// !!! https://github.com/labstack/echo/issues/1633
+
```
Comment (SunJinping-TRE-C) [inline framework/bootstrap/kernel.go:81]: here too.

File: framework/bootstrap/kernel.go (hunk 2 above)
Comment (tanvir-retailai) [reply framework/bootstrap/kernel.go:66]: Task for @yu-yk

File: framework/bootstrap/kernel.go (hunk 3 above)
Comment (tanvir-retailai) [reply framework/bootstrap/kernel.go:81]: Task for @yu-yk

File: framework/bootstrap/kernel.go (hunk 1 above)
Comment (tanvir-retailai) [reply framework/bootstrap/kernel.go:57]: This is your task :)

**Review by yu-yk — APPROVED** [review]

**Review by SunJinping-TRE-C — APPROVED** [review]
LGTM

**Pull request merged**